build error with `cabal new-build`

when calling `cabal new-build` on a fresh clone, the build fails. (ghc-8.6.4 on fedora)

It seems that this is due to incorrect version bounds.

<details><summary>log</summary>
<p>

```
$ cabal new-build
Resolving dependencies...
Build profile: -w ghc-8.6.4 -O1
In order, the following will be built (use -v for more details):
 - brittany-0.11.0.0 (lib) (first run)
 - cabal-helper-0.9.0.0 (lib:cabal-helper) (first run)
 - floskell-0.9.0 (lib) (first run)
 - lsp-test-0.5.1.3 (lib) (requires build)
 - ghc-mod-core-5.9.0.0 (lib) (first run)
 - hie-plugin-api-0.6.0.0 (lib) (first run)
 - ghc-mod-5.9.0.0 (lib:ghc-mod) (first run)
 - HaRe-0.8.4.1 (lib) (first run)
 - haskell-ide-engine-0.8.0.0 (lib) (first run)
 - haskell-ide-engine-0.8.0.0 (test:wrapper-test) (first run)
 - haskell-ide-engine-0.8.0.0 (exe:hie-wrapper) (first run)
 - haskell-ide-engine-0.8.0.0 (lib:hie-test-utils) (first run)
 - haskell-ide-engine-0.8.0.0 (exe:hie) (first run)
 - haskell-ide-engine-0.8.0.0 (test:unit-test) (first run)
 - haskell-ide-engine-0.8.0.0 (test:plugin-dispatcher-test) (first run)
 - haskell-ide-engine-0.8.0.0 (test:dispatcher-test) (first run)
 - haskell-ide-engine-0.8.0.0 (test:func-test) (first run)
Starting     lsp-test-0.5.1.3 (lib)
Configuring library for floskell-0.9.0..
Configuring library for brittany-0.11.0.0..
[1 of 1] Compiling Main             ($REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/cabal-helper-0.9.0.0/setup/setup.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/cabal-helper-0.9.0.0/setup/Main.o )
Linking$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/cabal-helper-0.9.0.0/setup/setup ...
Building     lsp-test-0.5.1.3 (lib)
Preprocessing library for brittany-0.11.0.0..
Building library for brittany-0.11.0.0..
Preprocessing library for floskell-0.9.0..
Building library for floskell-0.9.0..
[ 1 of 11] Compiling Floskell.Attoparsec ( src/Floskell/Attoparsec.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Attoparsec.o )
[ 1 of 30] Compiling Language.Haskell.Brittany.Internal.Prelude ( src/Language/Haskell/Brittany/Internal/Prelude.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Prelude.o )
[ 2 of 11] Compiling Floskell.Buffer  ( src/Floskell/Buffer.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Buffer.o )
[ 3 of 11] Compiling Floskell.Config  ( src/Floskell/Config.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Config.o )
[ 2 of 30] Compiling Language.Haskell.Brittany.Internal.PreludeUtils ( src/Language/Haskell/Brittany/Internal/PreludeUtils.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/PreludeUtils.o )
Configuring cabal-helper-0.9.0.0...
[ 3 of 30] Compiling Language.Haskell.Brittany.Internal.Obfuscation ( src/Language/Haskell/Brittany/Internal/Obfuscation.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Obfuscation.o )
[ 4 of 30] Compiling Language.Haskell.Brittany.Internal.Config.Types ( src/Language/Haskell/Brittany/Internal/Config/Types.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Config/Types.o )
Preprocessing library for cabal-helper-0.9.0.0..
Building library for cabal-helper-0.9.0.0..
[1 of 4] Compiling CabalHelper.Shared.InterfaceTypes ( src/CabalHelper/Shared/InterfaceTypes.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/cabal-helper-0.9.0.0/build/CabalHelper/Shared/InterfaceTypes.o )
[2 of 4] Compiling CabalHelper.Shared.Sandbox ( src/CabalHelper/Shared/Sandbox.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/cabal-helper-0.9.0.0/build/CabalHelper/Shared/Sandbox.o )
[3 of 4] Compiling Paths_cabal_helper ($REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/cabal-helper-0.9.0.0/build/autogen/Paths_cabal_helper.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/cabal-helper-0.9.0.0/build/Paths_cabal_helper.o )
[4 of 4] Compiling Distribution.Helper ( lib/Distribution/Helper.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/cabal-helper-0.9.0.0/build/Distribution/Helper.o )

lib/Distribution/Helper.hs:124:1: warning: [-Wunused-imports]
    The import of ‘Distribution.Simple.BuildPaths’ is redundant
      except perhaps to import instances from ‘Distribution.Simple.BuildPaths’
    To import instances alone, use: import Distribution.Simple.BuildPaths()
    |
124 | import           Distribution.Simple.BuildPaths    (exeExtension)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

lib/Distribution/Helper.hs:563:1: warning: [-Wunused-top-binds]
    Defined but not used: ‘isCabalHelperSourceDir’
    |
563 | isCabalHelperSourceDir dir =
    | ^^^^^^^^^^^^^^^^^^^^^^
[ 5 of 30] Compiling Language.Haskell.Brittany.Internal.Config.Types.Instances ( src/Language/Haskell/Brittany/Internal/Config/Types/Instances.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Config/Types/Instances.o )
[ 6 of 30] Compiling Language.Haskell.Brittany.Internal.Types ( src/Language/Haskell/Brittany/Internal/Types.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Types.o )
[ 7 of 30] Compiling Language.Haskell.Brittany.Internal.Utils ( src/Language/Haskell/Brittany/Internal/Utils.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Utils.o )
[ 8 of 30] Compiling Language.Haskell.Brittany.Internal.Transformations.Par ( src/Language/Haskell/Brittany/Internal/Transformations/Par.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Transformations/Par.o )
[ 9 of 30] Compiling Language.Haskell.Brittany.Internal.Transformations.Indent ( src/Language/Haskell/Brittany/Internal/Transformations/Indent.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Transformations/Indent.o )
[10 of 30] Compiling Language.Haskell.Brittany.Internal.Transformations.Floating ( src/Language/Haskell/Brittany/Internal/Transformations/Floating.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Transformations/Floating.o )
[11 of 30] Compiling Language.Haskell.Brittany.Internal.Transformations.Columns ( src/Language/Haskell/Brittany/Internal/Transformations/Columns.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Transformations/Columns.o )
[12 of 30] Compiling Language.Haskell.Brittany.Internal.Transformations.Alt ( src/Language/Haskell/Brittany/Internal/Transformations/Alt.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Transformations/Alt.o )
[13 of 30] Compiling Language.Haskell.Brittany.Internal.ExactPrintUtils ( src/Language/Haskell/Brittany/Internal/ExactPrintUtils.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/ExactPrintUtils.o )
[14 of 30] Compiling Language.Haskell.Brittany.Internal.LayouterBasics ( src/Language/Haskell/Brittany/Internal/LayouterBasics.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/LayouterBasics.o )
[ 4 of 11] Compiling Floskell.Fixities ( src/Floskell/Fixities.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Fixities.o )
[ 5 of 11] Compiling Floskell.Styles  ( src/Floskell/Styles.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Styles.o )
[ 6 of 11] Compiling Floskell.ConfigFile ( src/Floskell/ConfigFile.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/ConfigFile.o )
[15 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Type ( src/Language/Haskell/Brittany/Internal/Layouters/Type.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Type.o )
[ 7 of 11] Compiling Floskell.Types   ( src/Floskell/Types.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Types.o )
[ 8 of 11] Compiling Floskell.Printers ( src/Floskell/Printers.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Printers.o )
[ 9 of 11] Compiling Floskell.Pretty  ( src/Floskell/Pretty.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Pretty.o )
[16 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Stmt[boot] ( src/Language/Haskell/Brittany/Internal/Layouters/Stmt.hs-boot,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Stmt.o-boot )
[17 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.IE ( src/Language/Haskell/Brittany/Internal/Layouters/IE.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/IE.o )
[18 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Import ( src/Language/Haskell/Brittany/Internal/Layouters/Import.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Import.o )
[19 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Module ( src/Language/Haskell/Brittany/Internal/Layouters/Module.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Module.o )
[20 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Expr[boot] ( src/Language/Haskell/Brittany/Internal/Layouters/Expr.hs-boot,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Expr.o-boot )
[21 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Pattern ( src/Language/Haskell/Brittany/Internal/Layouters/Pattern.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Pattern.o )
[22 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Decl ( src/Language/Haskell/Brittany/Internal/Layouters/Decl.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Decl.o )
[23 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Stmt ( src/Language/Haskell/Brittany/Internal/Layouters/Stmt.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Stmt.o )
[24 of 30] Compiling Language.Haskell.Brittany.Internal.Layouters.Expr ( src/Language/Haskell/Brittany/Internal/Layouters/Expr.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Layouters/Expr.o )
[25 of 30] Compiling Language.Haskell.Brittany.Internal.Config ( src/Language/Haskell/Brittany/Internal/Config.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Config.o )
[26 of 30] Compiling Language.Haskell.Brittany.Internal.BackendUtils ( src/Language/Haskell/Brittany/Internal/BackendUtils.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/BackendUtils.o )
[27 of 30] Compiling Language.Haskell.Brittany.Internal.Backend ( src/Language/Haskell/Brittany/Internal/Backend.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal/Backend.o )
[10 of 11] Compiling Floskell.Comments ( src/Floskell/Comments.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell/Comments.o )
[28 of 30] Compiling Language.Haskell.Brittany.Internal ( src/Language/Haskell/Brittany/Internal.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany/Internal.o )
[11 of 11] Compiling Floskell         ( src/Floskell.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/floskell-0.9.0/build/Floskell.o )

src/Floskell.hs:32:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
32 | import           Data.Monoid
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[29 of 30] Compiling Language.Haskell.Brittany ( src/Language/Haskell/Brittany.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Language/Haskell/Brittany.o )
[30 of 30] Compiling Paths_brittany   ($REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/autogen/Paths_brittany.hs,$REPO/dist-newstyle/build/x86_64-linux/ghc-8.6.4/brittany-0.11.0.0/build/Paths_brittany.o )

Failed to build lsp-test-0.5.1.3.
Build log (
$HOME/.cabal/logs/ghc-8.6.4/lsp-test-0.5.1.3-09e8af3f25489fc3963d372ebb66f59bf47475d29f04a066ca55a5359b9be5aa.log
):
Configuring library for lsp-test-0.5.1.3..
Preprocessing library for lsp-test-0.5.1.3..
Building library for lsp-test-0.5.1.3..
[ 1 of 10] Compiling Language.Haskell.LSP.Test.Compat ( src/Language/Haskell/LSP/Test/Compat.hs, dist/build/Language/Haskell/LSP/Test/Compat.o )
[ 2 of 10] Compiling Language.Haskell.LSP.Test.Exceptions ( src/Language/Haskell/LSP/Test/Exceptions.hs, dist/build/Language/Haskell/LSP/Test/Exceptions.o )
[ 3 of 10] Compiling Language.Haskell.LSP.Test.Decoding ( src/Language/Haskell/LSP/Test/Decoding.hs, dist/build/Language/Haskell/LSP/Test/Decoding.o )

src/Language/Haskell/LSP/Test/Decoding.hs:134:9: error:
    Not in scope: data constructor ‘WindowProgressStart’
    |
134 |         WindowProgressStart            -> NotProgressStart $ fromJust $ decode bytes
    |         ^^^^^^^^^^^^^^^^^^^

src/Language/Haskell/LSP/Test/Decoding.hs:135:9: error:
    Not in scope: data constructor ‘WindowProgressReport’
    |
135 |         WindowProgressReport           -> NotProgressReport $ fromJust $ decode bytes
    |         ^^^^^^^^^^^^^^^^^^^^

src/Language/Haskell/LSP/Test/Decoding.hs:136:9: error:
    Not in scope: data constructor ‘WindowProgressDone’
    |
136 |         WindowProgressDone             -> NotProgressDone $ fromJust $ decode bytes
    |         ^^^^^^^^^^^^^^^^^^

src/Language/Haskell/LSP/Test/Decoding.hs:137:9: error:
    Not in scope: data constructor ‘WindowProgressCancel’
    |
137 |         WindowProgressCancel           -> NotProgressCancel $ fromJust $ decode bytes
    |         ^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build lsp-test-0.5.1.3 (which is required by test:func-test
from haskell-ide-engine-0.8.0.0). See the build log above for details.
```
</p>
</details>

--- a/haskell-ide-engine.cabal
+++ b/haskell-ide-engine.cabal
@@ -275,7 +275,7 @@ test-suite func-test
                      , data-default
                      , directory
                      , filepath
-                     , lsp-test >= 0.5.1.1 && < 0.5.2
+                     , lsp-test == 0.5.1.1
                      , haskell-ide-engine
                      , haskell-lsp-types >= 0.4
                      , hie-test-utils

